### ISSUE TITLE
Docker 2.2.0.0 won't uninstall

### ISSUE THREAD

**Issue by brianmaher84** [docker/for-win#5570]
### Expected behavior

I right click docker desktop in the start menu and click uninstall.  

A dialog appears saying it is uninstalling with text "Removing VM and images".

This dialog should progress and give me an option to close it when uninstall is comlete

### Actual behavior

I exit docker desktop and stop the service to ensure nothing is running.  I delete the .docker folder in my home directory.

I right click docker desktop in the start menu and click uninstall.  

A dialog appears saying it is uninstalling with text "Removing VM and images".

It is stuck at this phase of uninstall even after leaving it over night.

**Comment by mat007** [comment]
Thanks for your report, could you attach `%LOCALAPPDATA%\Docker\install-log.txt` please?

**Comment by brianmaher84** [comment]
[install-log.txt](https://github.com/docker/for-win/files/4103380/install-log.txt)

**Comment by mat007** [comment]
Thanks, looking into it!

**Comment by brianmaher84** [comment]
Could it be related to the reason I'm uninstalling?  At the moment docker won't start and can't be reset to factory defaults.

So I'm just trying to uninstall and reinstall to try to fix that issue when I found that it won't uninstall either.

**Comment by mat007** [comment]
It looks like you enabled WSL 2 and we’re suspecting it fails to uninstall because the service was stopped beforehand.
Something you can try is open `%APPDATA%\Docker\settings.json` and locate the `wslEngineEnabled` entry near the bottom of the file and set it to `false` then try again reset to factory default or uninstall.

**Comment by brianmaher84** [comment]
Sounds likely, I will try the fix tomorrow and revert back.

**Comment by brianmaher84** [comment]
> 
> 
> It looks like you enabled WSL 2 and we’re suspecting it fails to uninstall because the service was stopped beforehand.
> Something you can try is open `%APPDATA%\Docker\settings.json` and locate the `wslEngineEnabled` entry near the bottom of the file and set it to `false` then try again reset to factory default or uninstall.

Unfortunately the wslEngineEnabled setting is false.  I think I tried to enable it at one point but it was greyed out in the GUI.

My settings file is below.

`{
  "settingsVersion": 5,
  "autoStart": true,
  "checkForUpdates": true,
  "analyticsEnabled": true,
  "displayedWelcomeWhale": true,
  "displayed14393Deprecation": false,
  "displayRestartDialog": true,
  "displaySwitchWinLinContainers": true,
  "displaySwitchVersionPack": true,
  "latestBannerKey": "",
  "dataFolder": "C:\\ProgramData\\DockerDesktop\\vm-data",
  "memoryMiB": 2048,
  "swapMiB": 1024,
  "cpus": 2,
  "diskSizeMiB": 65536,
  "hypervCIDR": "10.0.75.0/28",
  "hypervIP": null,
  "vpnkitCIDR": "192.168.65.0/28",
  "proxyHttpMode": false,
  "overrideProxyHttp": "",
  "overrideProxyHttps": "",
  "overrideProxyExclude": "",
  "useDnsForwarder": true,
  "dns": "8.8.8.8",
  "kubernetesEnabled": false,
  "showKubernetesSystemContainers": false,
  "kubernetesInitialInstallPerformed": false,
  "versionPack": "default",
  "sharedDrives": {},
  "executableDate": "",
  "useWindowsContainers": false,
  "swarmFederationExplicitlyLoggedOut": false,
  "activeOrganizationName": null,
  "exposeDockerAPIOnTCP2375": false,
  "lifecycleTimeoutSeconds": 600,
  "customWslDistroDir": null,
  "wslEngineEnabled": false,
  "integratedWslDistros": null,
  "networkCIDR": null,
  "diskPath": null
}`

**Comment by brianmaher84** [comment]
I see this error now this morning:
Error response from daemon: open \\.\pipe\docker_engine_linux: The system cannot find the file specified.


   at Docker.Backend.DockerDaemonChecker.Check(Func`1 isDaemonProcessStillRunning)
   at Docker.Backend.BackendNamedPipeServer.<Run>b__8_2(Object[] args)
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass9_0.<Register>b__0(Object[] parameters)
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)

**Comment by mat007** [comment]
> I see this error now this morning when I tried to start docker:
Error response from daemon: open \\.\pipe\docker_engine_linux: The system cannot find the file specified.

Yeah, that’s basically the engine not ready to receive connections, which in itself doesn’t really explain the root cause.

**Comment by luzgit2020** [comment]
I am also seeing the same error -  Error response from daemon: open \.\pipe\docker_engine_linux: The system cannot find the file specified.

**Comment by brianmaher84** [comment]
Is there anyway to force uninstall docker so I can attempt re-installation.

I need to get it back up and running.

**Comment by mat007** [comment]
What happens if you simply try to re-install 2.1.0.5 on top?
If it says you’re up to date, use `regedit` to change the value of `Version` in `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Docker Desktop` to something very low (e.g. `1`) and re-try.
After installation you might want to force a reset to factory defaults to erase anything that might have been migrated to 2.2.0.0.

**Comment by brianmaher84** [comment]
Good news, I got 2.2.0.0 working.

First I removed all traces of it by running the script available here:
https://success.docker.com/article/how-to-completely-remove-docker-in-windows-10

Then I delete the docker files in C:\Program Data\ which the script didn't remove.

I followed the steps below to delete Docker Desktop from the add/remove programs.

I reinstalled but the issue persisted.

So I repeated the same removal steps then I disabled the "Windows Subsystem for Linux" in case it was causing an issue.  Rebooted the machine.

Reinstalled 2.2.0.0 and it came up no problem.  I suspect that just having the "Windows Subsystem for Linux" enabled might be causing some kind of issue.
